trieste/acquisition/optimizer.py

```diff
@@ -35,24 +34,83 @@
 AcquisitionOptimizer = Callable[[SP, AcquisitionFunction], TensorType]
 """
 Type alias for a function that returns the single point that maximizes an acquisition function over
-a search space. For a search space with points of shape S, and acquisition function with input shape
-[...] + S output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be [1] + S.
+a search space. For a search space with points of shape [D], and acquisition function with input
+shape [..., B, D] output shape [..., 1], the :const:`AcquisitionOptimizer` return shape should be
+[B, D].
 """
 
 
-@singledispatch
-def optimize(space: Box | DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+def automatic_optimizer_selector(
+    space: DiscreteSearchSpace, target_func: AcquisitionFunction
```

Comment:
if you make the type of `space` `Box | DiscreteSearchSpace` you won't need the exception below

Reply:
THe exception is more informative?

Reply:
why so? btw if this is just `DiscreteSearchSpace` it won't type check when called with `Box`